src/tests/core.py

```diff
@@ -274,6 +276,42 @@ def test_old_knownnodes_pickle(self):
         finally:
             cleanup(files=('knownnodes.dat',))
 
+    @staticmethod
+    def delete_address_from_addressbook(address):
+        """Clean up addressbook"""
+        sqlQuery('''delete from addressbook where address=?''', address)
+
+    def test_add_same_address_twice_in_addressbook(self):
+        """checking same address is added twice in addressbook"""
+        self.assertTrue(helper_addressbook.insert(label='test1', address=self.addr))
+        self.assertFalse(helper_addressbook.insert(label='test1', address=self.addr))
+        self.delete_address_from_addressbook(self.addr)
+
+    def test_is_address_present_in_addressbook(self):
+        """checking is address added in addressbook or not"""
+        helper_addressbook.insert(label='test1', address=self.addr)
+        queryreturn = sqlQuery('''select count(*) from addressbook where address=?''', self.addr)
+        self.assertTrue(bool(queryreturn[0][0]))
```

Comment:
I would do `assertEqual(return, 1)` in this case.

Reply:
here we are checking address is present in addressbook or not if anyhow count will show more then one it means the address is stored multiple times then in this case test will fail for assertEqual(return, 1) and it will pass for 
self.assertTrue(bool(queryreturn[0][0])) so i think we should keep it